domains/zeto/internal/zeto/fungible/handler_lock.go

```diff
@@ -233,70 +208,92 @@ func (h *lockHandler) Prepare(ctx context.Context, tx *types.ParsedTransaction,
 		return nil, i18n.NewError(ctx, msgs.MsgErrorUnmarshalProvingRes, err)
 	}
 
-	data, err := common.EncodeTransactionData(ctx, req.Transaction, types.ZetoTransactionData_V0)
+	inputSize := common.GetInputSize(len(req.InputStates))
+	inputs, err := utxosFromInputStates(ctx, req.InputStates, inputSize)
 	if err != nil {
-		return nil, i18n.NewError(ctx, msgs.MsgErrorEncodeTxData, err)
+		return nil, err
 	}
-	LockParams := map[string]interface{}{
-		"utxos":    decodedTransfer.Inputs,
-		"proof":    common.EncodeProof(proofRes.Proof),
-		"delegate": params.Delegate,
-		"data":     data,
+
+	lockedInfo, err := parseInfoStatesJson(req.InfoStates[0].StateDataJson)
+	if err != nil {
+		return nil, i18n.NewError(ctx, msgs.MsgErrorParseInfoStates, err)
 	}
-	paramsJSON, err := json.Marshal(LockParams)
+	unlockedOutputStates := req.OutputStates[:len(req.OutputStates)-int(lockedInfo.Size.Uint64())]
+	lockedOutputStates := req.OutputStates[len(req.OutputStates)-int(lockedInfo.Size.Uint64()):]
+
+	outputs, err := utxosFromOutputStates(ctx, unlockedOutputStates, inputSize)
 	if err != nil {
 		return nil, err
 	}
-	functionJSON, err := json.Marshal(lockStatesABI)
+	outputs = trimZeroUtxos(outputs)
+
+	lockedOutputs, err := utxosFromOutputStates(ctx, lockedOutputStates, inputSize)
 	if err != nil {
 		return nil, err
 	}
+	lockedOutputs = trimZeroUtxos(lockedOutputs)
 
-	return &prototk.PrepareTransactionResponse{
-		Transaction: &prototk.PreparedTransaction{
+	data, err := common.EncodeTransactionData(ctx, req.Transaction)
+	if err != nil {
+		return nil, i18n.NewError(ctx, msgs.MsgErrorEncodeTxData, err)
+	}
+	params := map[string]any{
+		"inputs":        inputs,
+		"outputs":       outputs,
+		"lockedOutputs": lockedOutputs,
+		"proof":         common.EncodeProof(proofRes.Proof),
+		"delegate":      tx.Params.(*types.LockParams).Delegate.String(),
+		"data":          data,
+	}
+	transferFunction := getLockABI(tx.DomainConfig.TokenName)
+	if common.IsNullifiersToken(tx.DomainConfig.TokenName) {
+		delete(params, "inputs")
```

Comment:
Rather than setting "inputs" and then deleting it, maybe we should only set it in an `else` case.